instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/RouteRefreshTest.kt

```diff
@@ -187,6 +195,44 @@ class RouteRefreshTest : BaseTest<EmptyTestActivity>(EmptyTestActivity::class.ja
         assertEquals(1180.651, refreshedDuration, 0.0001)
     }
 
+    @Test
+    fun routeRefreshesAfterCleanup() = sdkTest {
+        val routeOptions = generateRouteOptions(coordinates)
+        val routes = mapboxNavigation.requestRoutes(routeOptions)
+            .getSuccessfulResultOrThrowException()
+            .routes
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(routes)
+        mapboxNavigation.startTripSession()
+        stayOnInitialPosition()
+
+        waitForRouteToRefresh()
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(listOf())
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(routes)
+        waitForRouteToRefresh()
```

Comment:
can also check that refreshed route has expected annotations, incidents and id?

Reply:
The goal of this test is to verify that route refreshes doesn't stop working after cleaning up and setting a new route. The way separate routes is refreshed is well tested in [unit tests](https://github.com/mapbox/mapbox-navigation-android/blob/6413f57e6ddd9138ee6e51cca5dd9860d1785435/libnavigation-core/src/test/java/com/mapbox/navigation/core/routerefresh/RouteRefreshControllerTest.kt).

Do you see any aspect of behaviour which you can break without making at least 1 test red?

Reply:
we actually verify it in other tests, so can skip there. Could you rename the test to align the name and aim

Reply:
Does it look better now?